system/lib/pthread/library_pthread.c

```diff
@@ -185,7 +186,7 @@ static void _do_call(void* arg) {
   switch (q->functionEnum) {
     case EM_PROXIED_RESIZE_OFFSCREENCANVAS:
```

Comment:
Do we still need `EM_PROXIED_RESIZE_OFFSCREENCANVAS` at all?  Can `emscripten_set_offscreencanvas_size` be used at the call sites instead?

Happy to see this as a followup too.

Reply:
Good question. I looked at the usage of `EM_PROXIED_RESIZE_OFFSCREENCANVAS`, but it seems that it dispatches that call to a specific thread, which can't be done with the `__proxy: 'sync'` approach (AFAIK).
https://github.com/emscripten-core/emscripten/blob/f242a3e0fc87c29d66ccab0a0caa28871c631135/src/library_html5.js#L2502
https://github.com/emscripten-core/emscripten/blob/f242a3e0fc87c29d66ccab0a0caa28871c631135/src/library_pthread.js#L777-L782